src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -123,54 +123,69 @@ private static void AddChannelBindingToken(X509Certificate2 certificate, IntPtr
 
             private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
             {
-                List<X509Certificate2> otherCerts;
-                bool success;
-
                 using (SafeX509StoreCtxHandle storeCtx = new SafeX509StoreCtxHandle(storeCtxPtr, ownsHandle: false))
-                using (X509Chain chain = new X509Chain())
                 {
-                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
-
-                    IntPtr leafCertPtr = Interop.Crypto.X509StoreCtxGetTargetCert(storeCtx);
-
-                    if (IntPtr.Zero == leafCertPtr)
+                    // First use the default verification provided directly by OpenSSL.
```

Comment:
There are two assumptions here that we might want to codify (in comments):

1) That we don't have a notion of the Windows Disallowed store. (So users can't explicitly un-trust things that the system trusts).
2) That we're not doing CRL checking

If we had a Disallowed store then we'd need to ensure it was empty before doing this (or check after to make sure no Disallowed cert was involved).

If CRL-checking is enabled we'd potentially need to download the CRLs, which is part of our chainwalk logic.